install.ps1

```diff
@@ -13,81 +14,291 @@ param (
+            return $inputPath
+        } else {
+            Write-Host "Invalid path: $inputPath. Please enter a valid path."
+        }
     }
+}
 
-    Invoke-WebRequest -Uri $url -OutFile $env:temp/$filename -ErrorAction Stop
-    Expand-Archive -Path $env:temp/$filename -DestinationPath $InstallPath -Force -ErrorVariable ProcessError;
-    If ($ProcessError) {
-@"
----------------------------------------------------------------------------
-Whoops apparently we had an issue in unzipping the file, please check
-you have the right permission to do so and try to unzip manually the file.
-Currently we saved Dagger at your temp folder.
----------------------------------------------------------------------------
-"@
-        exit
-    } else {
-@"
+function Get-ProcessorArchitecture {
 
-Thank You for downloading Dagger!
+    $arch = $env:PROCESSOR_ARCHITECTURE
 
------------------------------------------------------
-Dagger has been saved at <YOUR HOME FOLDER>/dagger/
-Please add dagger.exe to your PATH in order to use it
-----------------------------------------------------
+    switch ($arch) {
+        "AMD64" {
+            return "amd64"
+        }
+        "ARM64" {
+            return "arm64"
+        }
+        "ARM" {
+            return "armv7"
+        }
+        default {
+            throw "Unsupported architecture: $arch"
+        }
+    }
+}
 
-"@
+function Get-ValidPath {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Message,
+        [Parameter(Mandatory=$true)]
+        [string]$Default
+    )
+
+    $path = $null
+    while ($null -eq $path) {
+        $inputPath = Read-Host -Prompt $Message
+
+        if ([string]::IsNullOrWhiteSpace($inputPath)) {
+            return $Default
+        } else {
+            if (Test-Path $inputPath -IsValid) {
+                return $inputPath
+            } else {
+                Write-Host "Invalid path: $inputPath. Please enter a valid path."
+            }
+        }
     }
+}
 
+function Get-DownloadUrl {
+    $fileName = Get-FileName
+    return "https://github.com/dagger/dagger/releases/download/v${DaggerVersion}/${fileName}"
 }
 
-function latest_version {
-    $response = Invoke-RestMethod 'http://dl.dagger.io/dagger/latest_version' -Method 'GET' -Body $body -ErrorAction SilentlyContinue -ErrorVariable DownloadError
-    If ($DownloadError) {
-@"
+# Used for interactive mode to get a true or false response from the user
+function Get-TrueFalse {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Message,
+        [Parameter(Mandatory=$true)]
+        [bool]$Default
+    )
+
+    $response = ""
+    while ($response -notmatch "^(y|n)$") {
+        $response = Read-Host -Prompt $Message
+        if ([string]::IsNullOrWhiteSpace($response)) {
+            $response = $Default ? "y" : "n"
+            break
+        }
+    }
+
+    return $response.StartsWith("y") ? $true : $false
+}
+
+function Find-LatestVersion {
+
```

Comment:
nit: empty line (and in several other places)

Reply:
Got rid of a lot of empty lines